python/plugins/processing/core/parameters.py

```diff
@@ -79,13 +80,25 @@ def setValue(self, obj):
         of parameter.
         """
         if obj is None:
-            if not self.optional:
+            if not self.optional and self.default is None:
                 return False
-            self.value = None
-            return True
+            elif self.default is None:
+                self.value = None
+                return True
+            else:
```

Comment:
I probably misunderstand the discussion linked to the commit 3472ac80d8bdd030090ce81cb686f3cfcbd913cd
I reproduce the way Processing worked:
* does not change value at the init
* apply default if None is passed to setValue

I propose this code for a review of each parameter.

@volaya what do you prefer ?